fairscale/optim/grad_scaler.py

```diff
@@ -38,36 +75,306 @@ def __init__(
         growth_interval: int = 2000,
         enabled: bool = True,
         process_group: Any = dist.group.WORLD,
-    ) -> None:
```

Comment:
Can we update the class doctring to account for the new changes?

Reply:
done